fpdf/drawing.py

```diff
@@ -0,0 +1,4207 @@
+    """
+    Produce a DeviceGray color from the given 8-bit gray value.
+
+    Args:
+        g (Number): gray color component. Must be in the interval [0, 255]. 0 is black,
+            255 is white.
+        a (Optional[Number]): alpha component. Must be `None` or in the interval
+            [0, 255]. 0 is fully transparent, 255 is fully opaque
+
+    Returns:
+        DeviceGray color representation.
+
+    Raises:
+        ValueError: if any components are not in their valid interval.
+    """
+    if a is not None:
+        a /= 255.0
+
+    return DeviceGray(g / 255.0, a)
+
+
+def cmyk8(c, m, y, k, a=None):
+    """
+    Produce a DeviceCMYK color from the given 8-bit CMYK values.
+
+    Args:
+        c (Number): red color component. Must be in the interval [0, 255].
+        m (Number): green color component. Must be in the interval [0, 255].
+        y (Number): blue color component. Must be in the interval [0, 255].
+        k (Number): blue color component. Must be in the interval [0, 255].
+        a (Optional[Number]): alpha component. Must be `None` or in the interval
+            [0, 255]. 0 is fully transparent, 255 is fully opaque
+
+    Returns:
+        DeviceCMYK color representation.
+
+    Raises:
+        ValueError: if any components are not in their valid interval.
+    """
+    if a is not None:
+        a /= 255.0
+
+    return DeviceCMYK(c / 255.0, m / 255.0, y / 255.0, k / 255.0, a)
+
+
+def color_from_hex_string(hexstr):
+    """
+    Parse an RGB color from a css-style 8-bit hexadecimal color string.
+
+    Args:
+        hexstr (str): of the form `#RGB`, `#RGBA`, `#RRGGBB`, or `#RRGGBBAA`. Must
+            include the leading octothorp. Forms omitting the alpha field are
+            interpreted as not specifying the opacity, so it will not be explicitly set.
+
+            An alpha value of `00` is fully transparent and `FF` is fully opaque.
+
+    Returns:
+        DeviceRGB representation of the color.
+    """
+    if not isinstance(hexstr, str):
+        raise TypeError(f"{hexstr} is not of type str")
+
+    if not hexstr.startswith("#"):
+        raise ValueError(f"{hexstr} does not start with #")
+
+    hlen = len(hexstr)
+
+    if hlen == 4:
+        return rgb8(*[int(char * 2, base=16) for char in hexstr[1:]], a=None)
+
+    if hlen == 5:
+        return rgb8(*[int(char * 2, base=16) for char in hexstr[1:]])
+
+    if hlen == 7:
+        return rgb8(
+            *[int(hexstr[idx : idx + 2], base=16) for idx in range(1, hlen, 2)], a=None
+        )
+
+    if hlen == 9:
+        return rgb8(*[int(hexstr[idx : idx + 2], base=16) for idx in range(1, hlen, 2)])
+
+    raise ValueError(f"{hexstr} could not be interpreted as a RGB(A) hex string")
+
+
+class Point(NamedTuple):
+    """
+    An x-y coordinate pair within the two-dimensional coordinate frame.
+    """
+
+    x: Number
+    """The abscissa of the point."""
+
+    y: Number
+    """The ordinate of the point."""
+
+    def render(self):
```

Comment:
Hmm, this is a reasonable suggestion, but I'd say that in my opinion the difference is that the `pdf_repr` method is something that ought to return serialized data able to be written directly to the PDF as a self-contained PDF primitive or collection of primitives. Since the `Point` requires additional info to give it meaning (for example, a trailing `m`, `l`, `c`, etc. to turn it into a drawing path component), I'd consider it distinct in purpose and leave it with the different name to maintain that distinction. That's a fairly fine distinction and not a hill I feel the need to die on, so if you think it ought to be changed, I'm willing to do that.

Reply:
No, keep it that way, I'm fine with it :)